### PR TITLE
[14.0] [MIG] account_payment_order: Migration to v14

### PR DESCRIPTION
**Pull request by CarlosRoca13**
cc @Tecnativa TT26418

This PR superseds https://github.com/OCA/bank-payment/pull/749 adding the suggested changes of @alexis-via and @StefanRijnhart because the author doesn't attend his PR since 24 Oct 2020.

Please review it, @pedrobaeza @joao-p-marques

### CODE REVIEW

File: account_payment_order/readme/USAGE.rst [hunk 1]
```diff
@@ -0,0 +1,7 @@
+You can create a Payment order via the menu Invoicing/Accounting > Payments > Payment Orders and then select the move lines to pay.
```
Comment (joao-p-marques) [inline account_payment_order/readme/USAGE.rst]:
```suggestion
You can create a Payment order via the menu Invoicing/Accounting > Vendors > Payment Orders and then select the move lines to pay.
```

File: account_payment_order/readme/USAGE.rst [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+You can create a Debit order via the menu Invoicing/Accounting > Payments > Debit Orders and then select the move lines to debit.
```
Comment (joao-p-marques) [inline account_payment_order/readme/USAGE.rst]:
```suggestion
You can create a Debit order via the menu Invoicing/Accounting > Customers > Debit Orders and then select the move lines to debit.
```

File: account_payment_order/readme/USAGE.rst (hunk 1 above)
Comment (joao-p-marques) [inline account_payment_order/readme/USAGE.rst]:
```suggestion
You can create a Payment order via the menu Invoicing/Accounting > Vendors > Payment Orders and then select the move lines to pay.
```

File: account_payment_order/readme/USAGE.rst [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+
+This module also adds a button *Add to Payment Order* on supplier invoices and a button *Add to Debit Order* on customer invoices.
```
Comment (joao-p-marques) [inline account_payment_order/readme/USAGE.rst]:
```suggestion
This module also adds an action *Add to Payment Order* on supplier invoices and *Add to Debit Order* on customer invoices.
```

File: account_payment_order/readme/USAGE.rst [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+
+You can print a Payment order via the menu Invoicing/Accounting > Payments > Payment Orders and then select the payment oder to print.
```
Comment (joao-p-marques) [inline account_payment_order/readme/USAGE.rst]:
```suggestion
You can print a Payment order via the menu Invoicing/Accounting > Vendors > Payment Orders and then select the payment oder to print.
```

**Review by joao-p-marques — COMMENTED** [review]
LGTM :+1: 
Just some comments to the README that can improve the documentation.

**Comment by CarlosRoca13** [comment]
@joao-p-marques Changes done!

**Review by joao-p-marques — APPROVED** [review]

**Review by victoralmau — APPROVED** [review]

File: account_payment_order/security/ir.model.access.csv [hunk 5]
```diff
@@ -4,3 +4,5 @@ access_account_payment_line,Full access on account.payment.line to Payment Manag
 access_bank_payment_line,Full access on bank.payment.line to Payment Manager,model_bank_payment_line,group_account_payment,1,1,1,1
 base.access_res_partner_bank_group_partner_manager,Full access on res.partner.bank to Account Payment group,base.model_res_partner_bank,group_account_payment,1,1,1,1
 base.access_res_bank_group_partner_manager,Full access on res.bank to Account Payment group,base.model_res_bank,group_account_payment,1,1,1,1
+access_account_payment_line_create,access_account_payment_line_create,model_account_payment_line_create,group_account_payment,1,1,1,1
```
Comment (pedrobaeza) [inline account_payment_order/security/ir.model.access.csv:7]: Is the button that launches this accessible only to the same group?

**Review by pedrobaeza — COMMENTED** [review]
Have you reused some work from the previous PR? If so, you have to at least add the contributor, and preserve commits if applicable

@alexis-via @StefanRijnhart are you going to review this?

File: account_payment_order/views/account_payment_order.xml [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+                        <page name="payment-lines" string="Transactions">
+                            <field
+                                name="payment_line_ids"
+                                context="{'default_payment_type': payment_type}"
+                            />
+                        </page>
+                        <page
+                            name="bank-payment-lines"
+                            string="Bank Payment Lines"
+                            invisible="True"
```
Comment (StefanRijnhart) [inline account_payment_order/views/account_payment_order.xml]: Thanks for adding the smart buttons! One of the benefits is that loading of the form of large orders (ours are > 20.000 lines) is faster when these large lists are not loaded by default. You chose to keep them here, I see, marked as invisible which defeats that benefit. Would you consider removing them entirely?

**Comment by CarlosRoca13** [comment]
@StefanRijnhart change done!

**Review by StefanRijnhart — APPROVED** [review]
Thanks! Tested and reviewed the changes with the earlier PR.

File: account_payment_order/security/ir.model.access.csv (hunk 5 above)
Comment (CarlosRoca13) [reply account_payment_order/security/ir.model.access.csv:7]: Yes, it is

**Comment by CarlosRoca13** [comment]
@pedrobaeza change on commits and Contributors done!

**Comment by pedrobaeza** [comment]
Great, thanks to all:

/ocabot merge nobump

**Pull request merged**